cfme/containers/node.py

```diff
@@ -95,14 +113,25 @@ def get_random_instances(cls, provider, count=1, appliance=None):
         return [collection.instantiate(obj.name, provider)
                 for obj in itertools.islice(node_list, count)]
 
+    @property
+    def exists(self):
```

Comment:
Since this is UI, shouldn't we check against the UI and not the backend?

Reply:
I don't think so, AFAIK you are not testing here, you are just checking if the node exists.
this is only my opinion - up to you.

Reply:
I think since we are modeling the UI here, we should keep it as is. I see the same being done in a couple other places as well. Examples:
https://github.com/ManageIQ/integration_tests/blob/597ce97bed0bb329d8630f0be540235f0f17f751/cfme/cloud/tenant.py#L227
https://github.com/ManageIQ/integration_tests/blob/597ce97bed0bb329d8630f0be540235f0f17f751/cfme/cloud/stack.py#L231

Reply:
Maybe you right. @psav what you think?

Reply:
```.exists``` here should check the UI
Methods against the models are related to the objects in MiQ therefore it does not make sense to check against the mgmt system. There are many places in the codebase right now where ```self.exists``` on an object checks either via REST or UI, but always MiQ. That's kinda why ```self.mgmt.exists``` is there, so that you can do a check of the management system.